toothpick-sample/src/main/java/com/example/toothpick/activity/AdvancedBackpackItemsActivity.kt

```diff
@@ -0,0 +1,116 @@
+package com.example.toothpick.activity
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Button
+import androidx.appcompat.app.AppCompatActivity
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.toothpick.R
+import com.example.toothpick.adapter.BackpackAdapter
+import com.example.toothpick.adapter.IBackpackAdapter
+import com.example.toothpick.annotation.ApplicationScope
+import com.example.toothpick.annotation.ViewModelScope
+import com.example.toothpick.helper.NotificationHelper
+import com.example.toothpick.model.Backpack
+import com.example.toothpick.viewmodel.BackpackViewModel
+import toothpick.Scope
+import toothpick.ktp.KTP
+import toothpick.ktp.binding.bind
+import toothpick.ktp.binding.module
+import toothpick.ktp.binding.toClass
+import toothpick.ktp.delegate.inject
+import toothpick.ktp.delegate.lazy
+import toothpick.smoothie.lifecycle.closeOnDestroy
+import toothpick.smoothie.viewmodel.closeOnViewModelCleared
+import toothpick.smoothie.viewmodel.installViewModelBinding
+
+/**
+ * Advanced version of the BackpackItemsActivity.
+ *
+ * Here we retain the backpack on configuration changes using the ViewModel Scope.
```

Comment:
In this example, the backpack is retained on configuration changes as it belongs to the view model scope which follows the lifecycle of view model instances: when an instance is destroyed, and later recreated, the scope remains unchanged and the backpack instance will be the same.